transform_encoder.py

```diff
@@ -39,7 +39,7 @@ def __init__(
         max_length: Optional[int] = None,
         embedding_fn_name: str = '__call__',
         device: str = 'cpu',
-        num_threads: int = 1,
+        num_threads: int = 8,
```

Comment:
the default should be kept as 1. Our parallelization model is based on multiprocessing and sort of an Actor model. Adding multithreading by default may have a lot of wrong consequences

Reply:
changed to use 1 as the default number of threads